web/modules/custom/custom_weather/custom_weather.module

```diff
@@ -3,24 +3,21 @@ function custom_weather_block_info() {
   $blocks = array();
   $blocks['custom_weather_block'] = array(
     'info' => t('Custom Weather Block'),
-    'cache' => DRUPAL_CACHE_PER_ROLE,
   );
   return $blocks;
 }
 
-function custom_weather_block_view($delta = '') {
-  $block = array();
-  switch ($delta) {
-    case 'custom_weather_block':
-      $block['subject'] = t('Weather Forecast');
-      $block['content'] = _custom_weather_block_content();
-      break;
-  }
-  return $block;
-}
-
```

Comment:
I`m not sure but maybe you should use  - use Drupal\block\Entity\Block?  And you need to fix coding standards.